src/ipc/ipc4/helper.c

```diff
@@ -787,6 +787,23 @@ int ipc4_add_comp_dev(struct comp_dev *dev)
 	/* add new component to the list */
 	list_item_append(&icd->list, &ipc->comp_list);
 
+	/* If new component added to the pipeline is not fitted for deep buffering we need to
+	 * inform pipeline it cannot work on long periods.
+	 */
+	if (!dev->deep_buffering) {
+		if (!dev->pipeline) {
+			const uint32_t pipe_id = dev->ipc_config.pipeline_id;
+			struct ipc_comp_dev *ipc_comp =
+				ipc_get_comp_by_ppl_id(ipc,
+						       COMP_TYPE_PIPELINE,
+						       pipe_id);
```

Comment:
are we sure, that `ipc_comp` cannot be `NULL` here? `pipe_id` comes from an IPC... BTW, looks like `ipc4_add_comp_dev()` can be `static`